src/App.tsx

```diff
@@ -1,223 +1,178 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useCallback,
+  useContext, useEffect, useMemo, useRef, useState,
+} from 'react';
+import classNames from 'classnames';
+import {
+  createTodo, getTodos, deleteTodo, updateTodo,
+} from './api/todos';
 import { AuthContext } from './components/Auth/AuthContext';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { Todo } from './types/Todo';
+import { Error } from './components/Error';
+
+enum FilterType {
+  ALL = 'all',
+  ACTIVE = 'active',
+  COMPLETED = 'completed',
+}
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState('');
+  const [filter, setFilter] = useState<string>(FilterType.ALL);
+  const [changeAllTodos, setChangeAllTodos] = useState(true);
+  const [errorType, setError] = useState('');
+
+  const fetchTodo = async () => {
+    if (query.trim().length === 0) {
+      setError('Type something');
+      setTimeout(() => {
+        setError('');
+      }, 3000);
```

Comment:
I see that you clear error after 3 seconds in 3 places. How about writing helper function to show some error and clear it after x seconds? Than you call it in these 3 places instead of repetitive code.

Reply:
`const showError = (text: string, hideTimeout: number) => { implementation here... }`